calico-vpp-agent/watchers/net_watcher.go

```diff
@@ -0,0 +1,210 @@
+// Copyright (C) 2021 Cisco Systems Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package watchers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/projectcalico/vpp-dataplane/calico-vpp-agent/common"
+	networkv3 "github.com/projectcalico/vpp-dataplane/calico-vpp-agent/network"
+	"github.com/projectcalico/vpp-dataplane/vpplink"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/tomb.v2"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type VRF struct {
+	Tables [2]uint32 // one for ipv4, one for ipv6
+}
+
+type NetworkDefinition struct {
+	VRF               VRF
+	Vni               uint32
+	Name              string
+	LoopbackSwIfIndex uint32
+	Range             string
+}
+
+type NetWatcher struct {
+	log                *logrus.Entry
+	vpp                *vpplink.VppLink
+	client             client.WithWatch
+	stop               chan struct{}
+	networkDefinitions map[string]*NetworkDefinition
+}
+
+func NewNetWatcher(vpp *vpplink.VppLink, log *logrus.Entry) *NetWatcher {
+	kubernetesClient, err := NewClient(10 * time.Second, []func(s *runtime.Scheme) error{networkv3.AddToScheme})
+	if err != nil {
+		panic(fmt.Errorf("failed instantiating kubernetes client: %v", err))
+	}
+	w := NetWatcher{
+		log:                log,
+		vpp:                vpp,
+		client:             *kubernetesClient,
+		stop:               make(chan struct{}),
+		networkDefinitions: make(map[string]*NetworkDefinition),
+	}
+	return &w
+}
+
+func (w *NetWatcher) WatchNetworks(t *tomb.Tomb) error {
+	w.log.Infof("Net watcher starts")
+	netList := &networkv3.NetworkList{}
+	err := w.client.List(context.Background(), netList, &client.ListOptions{})
+	if err != nil {
+		return err
+	}
+	networks := make(map[string]*NetworkDefinition)
+	for _, net := range netList.Items {
+		netDef, err := w.OnNetAdded(&net)
+		if err != nil {
+			w.log.Error(err)
+			return err
+		}
+		networks[netDef.Name] = netDef
+	}
+	common.SendEvent(common.CalicoVppEvent{
+		Type: common.NetsSynced,
+		New:  networks,
+	})
+
+restart:
```

Comment:
`for {}` ?

Reply:
for is inside `w.watching(watcher)`

Reply:
Sorry I wasn't clear 😄 
I meant having a `for {}` loop instead of
````golang
restart:
some_logic()
goto restart
````